test/test_pytree.py

```diff
@@ -59,6 +60,33 @@ def run_test(tup):
         run_test((1., 2))
         run_test((torch.tensor([1., 2]), 2, 10, 9, 11))
 
+    def test_flatten_unflatten_namedtuple(self):
+        Point = namedtuple('Point', ['x', 'y'])
+
+        def run_test(tup):
+            expected_spec = TreeSpec(namedtuple, Point, [LeafSpec() for _ in tup])
+            values, treespec = tree_flatten(tup)
+            self.assertTrue(isinstance(values, list))
+            self.assertEqual(values, list(tup))
+            self.assertEqual(treespec, expected_spec)
+
+            unflattened = tree_unflatten(values, treespec)
+            self.assertEqual(unflattened, tup)
```

Comment:
How deep is assertEqual doing for nested structures?

Reply:
I can't find the source code responsible for this, but by some testing, it looks like the answer is pretty deep:
```
(Pdb) self.assertEqual((1, 2), (2, 3))
*** AssertionError: False is not true : Scalars failed to compare as equal! 1 != 2
(Pdb) self.assertEqual((torch.tensor(3),), (torch.tensor(4),))
*** AssertionError: False is not true : Tensors failed to compare as equal!Found 1 different element(s)
 (out of 1), with the greatest difference of 1 (3 vs. 4) occuring at index 0.
 (Pdb) self.assertEqual(([torch.tensor(3)],), ([torch.tensor(4)],))
*** AssertionError: False is not true : Tensors failed to compare as equal!Found 1 different element(s)
 (out of 1), with the greatest difference of 1 (3 vs. 4) occuring at index 0.
 ```